tools/gfx/d3d12/render-d3d12.h

```diff
@@ -1,11 +1,1799 @@
 // render-d3d12.h
 #pragma once
 
+#include "../command-encoder-com-forward.h"
+#include "../d3d/d3d-swapchain.h"
+#include "../mutable-shader-object.h"
 #include "../renderer-shared.h"
+#include "../simple-render-pass-layout.h"
+#include "../transient-resource-heap-base.h"
+#include "core/slang-basic.h"
+#include "core/slang-blob.h"
+#include "core/slang-chunked-list.h"
+#include "descriptor-heap-d3d12.h"
+#include "resource-d3d12.h"
+
+#define _CRT_SECURE_NO_WARNINGS
+#define WIN32_LEAN_AND_MEAN
+#define NOMINMAX
+#include <Windows.h>
+#undef WIN32_LEAN_AND_MEAN
+#undef NOMINMAX
+
+#include <d3d12.h>
+#include <dxgi1_4.h>
+
+#ifndef __ID3D12GraphicsCommandList1_FWD_DEFINED__
+// If can't find a definition of CommandList1, just use an empty definition
+struct ID3D12GraphicsCommandList1
+{};
+#endif
 
 namespace gfx
 {
+namespace d3d12
+{
+
+using namespace Slang;
+
+// Define function pointer types for PIX library.
+typedef HRESULT(WINAPI* PFN_BeginEventOnCommandList)(
+    ID3D12GraphicsCommandList* commandList, UINT64 color, _In_ PCSTR formatString);
+typedef HRESULT(WINAPI* PFN_EndEventOnCommandList)(ID3D12GraphicsCommandList* commandList);
+
+static const Int kMaxNumRenderFrames = 4;
+static const Int kMaxRTVCount = 8;
+
+struct D3D12DeviceInfo
```

Comment:
Does it need D3D12 prefix when in the d3d12 namespace?

Reply:
There is already a DeviceInfo type in `gfx` namespace, perhaps I should just rename it to something else.